CMakeLists.txt

```diff
@@ -55,33 +62,51 @@ target_link_libraries(sxg
 )
 
 ########################################
-## Testing
+## Installing
 ########################################
 
-enable_testing()
-file(COPY tests/testdata DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
-
-ExternalProject_Add(
-  gtest
-  GIT_REPOSITORY https://github.com/google/googletest
-  GIT_TAG master
-  SOURCE_DIR ${PROJECT_BINARY_DIR}/third_party/gtest
-  BINARY_DIR ${PROJECT_BINARY_DIR}/gtest
-  INSTALL_COMMAND ""
-)
-add_library(libgtest IMPORTED STATIC GLOBAL)
-add_dependencies(libgtest gtest)
-set_target_properties(libgtest PROPERTIES
-    "IMPORTED_LOCATION" "${PROJECT_BINARY_DIR}/gtest/lib/libgtest.a"
-    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
+set_target_properties(sxg PROPERTIES PUBLIC_HEADER "${HEADERS}")
+INSTALL(
+  TARGETS sxg
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  COMPONENT library
 )
-add_library(libgtest_main IMPORTED STATIC GLOBAL)
-add_dependencies(libgtest_main gtest)
-set_target_properties(libgtest_main PROPERTIES
-    "IMPORTED_LOCATION" "${PROJECT_BINARY_DIR}/gtest/lib/libgtest_main.a"
-    "IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
+INSTALL(
+  DIRECTORY ${CMAKE_SOURCE_DIR}/include/
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_PREFIX}
+  FILES_MATCHING PATTERN "*.h*"
 )
-set(GTEST_INCLUDE ${PROJECT_BINARY_DIR}/third_party/gtest/googletest/include)
+
+########################################
+## Testing
+########################################
+if (NOT SKIP_TEST)
```

Comment:
Should there be a space after `if` as in this line (because it's "if") or should there not as in Lines 23 & 26 (because it's just a function)?

Reply:
OK, made it uniform.